kuma/javascript/src/ga-provider.jsx

```diff
@@ -6,25 +6,35 @@ export type GAFunction = (...any) => void;
 
 export const CATEGORY_MONTHLY_PAYMENTS = 'monthly payments';
 
-const GA_QUERY_KEY = 'ga';
+const GA_SESSION_STORAGE_KEY = 'ga';
 
-const QUERY_PARAM_GA_DATA = {
-    'subscription-success': {
-        hitType: 'event',
-        eventCategory: CATEGORY_MONTHLY_PAYMENTS,
-        eventAction: 'successful subscription',
-        eventLabel: 'subscription-landing-page',
-    },
-    'banner-cta': {
-        hitType: 'event',
-        eventCategory: CATEGORY_MONTHLY_PAYMENTS,
-        eventAction: 'subscribe intent',
-        eventLabel: 'banner',
-    },
-};
+function getPostponedEvents() {
+    try {
+        return JSON.parse(
+            sessionStorage.getItem(GA_SESSION_STORAGE_KEY) || JSON.stringify([])
+        );
+    } catch (e) {
+        // No sessionStorage support
+        return [];
+    }
+}
 
-export function gaQuery(id: $Keys<typeof QUERY_PARAM_GA_DATA>) {
-    return GA_QUERY_KEY + '=' + id;
+/**
+ * Saves given events into sessionStorage so that they are sent once the next
+ * page has loaded. This should be used for events that need to be sent without
+ * delaying navigation to a new page (which would cancel pending network
+ * requests).
+ */
+export function gaSendOnNextPage(newEvents: any[]) {
+    try {
+        const events = getPostponedEvents();
+        sessionStorage.setItem(
+            GA_SESSION_STORAGE_KEY,
+            JSON.stringify(events.concat(newEvents))
+        );
+    } catch (e) {
+        // No sessionStorage support
```

Comment:
It's a nit but now you're conflating the forgiveness of `sessionStorage.setItem` with a bunch of other things. For example, `const events = getPostponedEvents();` could crash. The `JSON.stringify(` could fail. Or the `events.concat(` could fail. 
I think it ought to be:
```javascript
export function gaSendOnNextPage(newEvents: any[]) {
    const events = getPostponedEvents();
    const value = JSON.stringify(events.concat(newEvents));
    try {
        sessionStorage.setItem(GA_SESSION_STORAGE_KEY, value);
    } catch (e) {
        // No sessionStorage support
    }
}
```